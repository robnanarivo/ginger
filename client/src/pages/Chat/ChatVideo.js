import React from 'react';
import { Grid, Typography } from '@mui/material';
import ReactPlayer from 'react-player';

const ChatVideo = ({
  sender,
  url,
  timestamp,
  isSelf,
}) => {
  let align = 'left';
  if (isSelf) {
    align = 'right';
  }
  return (
    <Grid container sx={{ p: 2 }}>
      <Grid item xs={12} align={align}>
        <Typography sx={{ fontSize: 14, color: 'gray' }}>{sender}</Typography>
        <ReactPlayer url={url} width="500px" controls />
        <Typography sx={{ fontSize: 14, color: 'gray' }}>{timestamp}</Typography>
      </Grid>
    </Grid>
  );
};

export default ChatVideo;
