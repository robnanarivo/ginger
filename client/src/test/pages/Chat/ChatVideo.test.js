import React from 'react';
import renderer from 'react-test-renderer';
import ChatVideo from '../../../pages/Chat/ChatVideo';
import 'jest-styled-components';

describe('ChatVideo unit test', () => {
  it('ChatVideo Snapshot test', () => {
    const tree = renderer
      .create(<ChatVideo sender="You" isSelf timestamp="20211202" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
