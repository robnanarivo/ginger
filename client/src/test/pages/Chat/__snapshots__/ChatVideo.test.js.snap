// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatVideo unit test ChatVideo Snapshot test 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-1fkyx8n-MuiGrid-root"
>
  <div
    align="right"
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-vvr8k2-MuiTypography-root"
    >
      You
    </p>
    <div
      style={
        Object {
          "height": "360px",
          "width": "500px",
        }
      }
    />
    <p
      className="MuiTypography-root MuiTypography-body1 css-vvr8k2-MuiTypography-root"
    >
      20211202
    </p>
  </div>
</div>
`;
